### PR TITLE
[14.0][MIG] shopinvader_delivery_state

### PR DESCRIPTION
**Pull request by mmequignon**
Forward port of https://github.com/shopinvader/odoo-shopinvader/pull/1008

### CODE REVIEW

**Comment by simahawk** [comment]
pls rewrite the migration commit

File: shopinvader_delivery_state/hooks.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger(__name__)
+
+
+def post_init_hook(cr, registry):
+    _logger.info("Recompute sale.order.shopinvader_state")
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    env["sale.order"].search([])._compute_shopinvader_state()
```
Comment (i-vyshnevska) [inline shopinvader_delivery_state/hooks.py:14]: this can be a performance bottleneck

**Review by i-vyshnevska — APPROVED** [review]

File: shopinvader_delivery_state/hooks.py (hunk 1 above)
Comment (simahawk) [reply shopinvader_delivery_state/hooks.py:14]: Is not a bottleneck as it runs on install only. If you don't run this, product state will be incorrect.

**Comment by simahawk** [comment]
@mmequignon ping

**Review by simahawk — APPROVED** [review]

**Pull request merged**